script.js

```diff
@@ -1,12 +1,375 @@
+        imageElem.src = 'https://dozenpixels.com/static/img/blog/coming-soon.png'
+    } else {
+        imageElem.src = episode.image.medium
+    }
+
+    divElement.appendChild(imageElem)
+        // Adding a paragraph to it as a summary of the episode
+    let paragraph = createHtmlElement('P')
+    paragraph.className = 'episode-summary'
+    episode.summary === null || episode.summary === '' ?
+        (paragraph.innerText = ' The summary of this episode is coming soon.') :
+        (paragraph.innerHTML = episode.summary)
+
+    divElement.appendChild(paragraph)
+}
+
+// Search by Listening for keystroke events
+searchInput.addEventListener('keyup', searchFunc)
+    // filtering  the search
+function searchFunc() {
+    if (episodeSelectList.style.display === '') {
+        searchEpisode()
+    } else {
+        searchShow()
+    }
+}
+
+// a function Loading the episode select menu list
+function selectInputLoad(episodeList) {
+    //Create and append the options of select input
+    let option1 = createHtmlElement('option')
+    option1.value = 'All Episodes'
+    option1.text = 'All Episodes'
+    episodeSelectList.appendChild(option1)
+    for (let index = 0; index < episodeList.length; index++) {
+        let option = createHtmlElement('option')
+        option.text =
+            'S' +
+            episodeList[index]['season'].toString().padStart(2, '0') +
+            'E' +
+            episodeList[index]['number'].toString().padStart(2, '0') +
+            ' - ' +
+            episodeList[index].name
+        episodeSelectList.appendChild(option)
+    }
+}
+
+function searchEpisode() {
+    let epiListLength
+    let searchCount
+    let searchKey
+    if (searchInput.value === '') {
+        if (episodeSelectList.value.substr(0, 6) === 'All Ep') {
+            searchKey = ''
+        } else {
+            searchKey = episodeSelectList.value.substr(0, 6).toLowerCase()
+        }
+    } else {
+        searchKey = searchInput.value.toLowerCase()
+    }
+    searchCount = 0
+    let arrayEpisodes = Array.from(document.querySelectorAll('.episodeContainer'))
+    epiListLength = arrayEpisodes.length
+    arrayEpisodes.forEach((show) => {
+        let textInfo = show.innerText.toLowerCase()
+        if (textInfo.indexOf(searchKey) > -1) {
+            show.style.display = ''
+            searchCount += 1
+        } else {
+            show.style.display = 'none'
+        }
+    })
+    if (searchCount == 1) {
+        searchLabel.innerHTML = `Displaying ${searchCount}/${epiListLength} episode`
+    } else {
+        searchLabel.innerHTML = `Displaying ${searchCount}/${epiListLength} episodes`
+    }
+}
+
+function searchShow() {
+    let epiListLength
+    let searchCount
+    let searchKey
+    searchKey = searchInput.value.toLowerCase()
+    searchCount = 0
+
+    let showsArray = Array.from(document.querySelectorAll('.show-card'))
+    showsListLength = showsArray.length
+    showsArray.forEach((show) => {
```

Comment:
nice. there is another way where you can work here straight with the data not the dom, and using filter